examples/routing_example/public/is_error.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export function isError<T>(error: T | Error): error is Error {
+  return (error as Error).message !== undefined;
```

Comment:
Probably minor, but having a typeguard for errors based on a `message` property seems dangerous. We do expose the `HttpFetchError` type from `core`. `instanceof HttpFetchError` may be better of a check?

Reply:
Yes it is, it actually bit me too when at one point - the body correctly included `message`.  The other option is to stop doing `catch (e) { return e }` and let the errors throw.  The problem with that is that it's easy for the consumer to not handle the possibility of an error being thrown.  Having typescript return `Response | Error` forces the consumer to handle this possibility. 

Typescript needs, like java has (last I checked like 12 years ago), a `throws E` for certain functions to force the consumers to handle it, or get a ts error.

Reply:
There is an open issue for like 5 years now: https://github.com/microsoft/TypeScript/issues/13219. But the TS core team isn't really open on this subject.